add simple implement of WebView for win32

summerinsects opened the pull request:
Is it being implemented by official developing team? If yes, ignore it

This implement contains some simple functions such as loadUrl, loadFile,
evaluateJS
Other more functions have not been implemented yet
My environment is win8.1 vs2013, and it works well

Commit:
Merge pull request #1 from cocos2d/v3

update

Commit:
Merge pull request #2 from wohaaitinciu/v3

update

Commit:
add simple implement of WebView for win32

This implement contains some simple functions such as loadUrl, loadFile,
evaluateJS
Other more functions have not been implemented yet
My environment is win8.1 vs2013, and it works well

minggo commented:
Thanks.
No, we are not doing right now.

Will you continue to implement other functions?

summerinsects commented:
@minggo I will try my best.

minggo commented:
Thanks.

minggo commented:
You should exclude `UIWebViewImpl_win32.cpp` to be compiled on linux.
You can do it in two ways:
- modify `cocos/CMakeLists.txt` to exclude `UIWebViewImpl_win32.cpp`
- make `UIWebViewImpl_win32.cpp` can be included in all platforms, just add macros at begin and end like this

``` c++
// UIWebViewImpl_win32.cpp
#if CC_TARGET_PLATFROM == CC_PLATFORM_WINDOWS
...
#endif
```

summerinsects commented:
OK

Commit:
Update UIWebViewImpl_win32.cpp

Commit:
Update UIWebViewImpl_win32.h

minggo commented on cocos/ui/UIWebViewImpl_win32.cpp:
```diff
@@ -0,0 +1,417 @@
+            , m_bCanGoBack(true)
+            , m_bCanGoForward(true)
+            , m_dwCookie(0)
+            , m_ulRef(0)
```
Please use _xxx as member variables to obey cocos2d-x programming guide.

minggo commented on cocos/ui/UIWebViewImpl_win32.cpp:
```diff
@@ -0,0 +1,417 @@
+
+        bool createWebView()
+        {
+            bool bRet = false;
```
Please don't use Hungary style.
bRet -> ret

summerinsects commented:
These codes is form an old win32 project.
OK, I'll modify them later

minggo commented:
Thanks.

summerinsects commented:
@minggo 
Cound I pass nullptr to C-style win32 api functions, such as GetModuleHandle,
or initialize some handle with nullptr, such as HWND,
or compare handle with nullptr to check whether some function call has no error?

minggo commented:
I am not sure. I haven't noticed that it is win32 API, so use NULL is ok.

summerinsects commented:
Then, I keep using NULL when calling win32 APIs or doing some operation with handle?

minggo commented:
Yep, using NULL for win32 API.

Commit:
Update UIWebViewImpl_win32.cpp

Commit:
add more implement

1. add more implement: loadHTMLString canGoBack canGoForward
2. Use 3 callback Functions of webview

Commit:
fix conflicts

summerinsects commented:
I have added more implementation.
Since I have no IOS environment, I built an apk, and studied the 3 callbacks.
However, due to platform differences, there should be many differences when callback is called

Commit:
add onJsCallback

minggo commented:
> Since I have no IOS environment, I built an apk, and studied the 3 callbacks.
> However, due to platform differences, there should be many differences when callback is called

You are implementing for windows, is there any problem related to iOS?

summerinsects commented:
I want to make the platform differences as small as possible

minggo commented:
Got it.
Is this PR ready to be merged?

summerinsects commented:
Yes

Commit:
Modify `std::string' to `const std::string &' as issues 7958 said

Commit:
Merge branch 'cocos2d-v3' into loc_ver

Commit:
Update UIWebView.cpp

Commit:
Update UIWebView.h

Commit:
Update UIWebViewImpl_iOS.mm

Commit:
Update UIWebViewImpl_win32.cpp

Commit:
Update UIWebViewImpl_win32.h

summerinsects commented:
@minggo 
It seems that my version is too old. Should I recheckout and PR again?

minggo commented:
You are appreciated if you recheckout and PR again.
And please add files into VS project and make the test case work.
